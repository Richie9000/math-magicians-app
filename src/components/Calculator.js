/* eslint-disable */ 
import React from 'react';
import calculate from './logic/calculate';

class Calculator extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      total: null,
      next: null,
      operation: null,
    };
  }

  UpdateObj = (event) => {
    const val = event.target.value;
    const objstate = this.state;
    const newState = calculate(objstate, val);
    this.setState(newState);
    const calculatorDisplay = document.querySelector('.calculator-display');
    if (newState.total === null && newState.next === null && newState.operation === null) {
      calculatorDisplay.textContent = '0';
    } else if (newState.next !== null) {
      calculatorDisplay.textContent = newState.next
    } else if (newState.next === null && newState.total !== null) {
      calculatorDisplay.textContent = newState.total
    }
    console.log(newState);
  }

  render() {
    return (
      <div className="calculator">
        <div className="calculator-display">0</div>
        <div className="calculator-keypad">
          <div className="input-keys">
            <div className="function-keys">
              <button className="calculator-key key-clear" value="AC" type="button" onClick={this.UpdateObj}>AC</button>
              <button className="calculator-key key-sign" value="+/-" type="button" onClick={this.UpdateObj}>+/-</button>
              <button className="calculator-key key-percent" value="%" type="button" onClick={this.UpdateObj}>%</button>
            </div>
            <div className="digit-keys">
              <button className="calculator-key key-0" value="0" type="button" onClick={this.UpdateObj}>0</button>
              <button className="calculator-key key-dot" value="." type="button" onClick={this.UpdateObj}>●</button>
              <button className="calculator-key key-1" value="1" type="button" onClick={this.UpdateObj}>1</button>
              <button className="calculator-key key-2" value="2" type="button" onClick={this.UpdateObj}>2</button>
              <button className="calculator-key key-3" value="3" type="button" onClick={this.UpdateObj}>3</button>
              <button className="calculator-key key-4" value="4" type="button" onClick={this.UpdateObj}>4</button>
              <button className="calculator-key key-5" value="5" type="button" onClick={this.UpdateObj}>5</button>
              <button className="calculator-key key-6" value="6" type="button" onClick={this.UpdateObj}>6</button>
              <button className="calculator-key key-7" value="7" type="button" onClick={this.UpdateObj}>7</button>
              <button className="calculator-key key-8" value="8" type="button" onClick={this.UpdateObj}>8</button>
              <button className="calculator-key key-9" value="9" type="button" onClick={this.UpdateObj}>9</button>
            </div>
          </div>
          <div className="operator-keys">
            <button className="calculator-key key-add" value="+" type="button" onClick={this.UpdateObj}>+</button>
            <button className="calculator-key key-substract" value="-" type="button" onClick={this.UpdateObj}>-</button>
            <button className="calculator-key key-divide" value="÷" type="button" onClick={this.UpdateObj}>÷</button>
            <button className="calculator-key key-multiply" value="x" type="button" onClick={this.UpdateObj}>*</button>
            <button className="calculator-key key-equals" value="=" type="button" onClick={this.UpdateObj}>=</button>
          </div>
        </div>
      </div>
    );
  }
}

export default Calculator;
