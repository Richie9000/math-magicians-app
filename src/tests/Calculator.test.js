import React from 'react';
import { render, screen } from '@testing-library/react';
import Calculator from '../components/Calculator';
import '@testing-library/jest-dom';

describe('App', () => {
  test('renders "AC" button in Calculator component', () => {
    render(<Calculator />);
    expect(screen.getByText('AC')).toBeInTheDocument();
  });

  test('renders "=" button in Calculator component', () => {
    render(<Calculator />);
    expect(screen.getByText('=')).toBeInTheDocument();
  });

  test('Calculator snapshot test', () => {
    const myrender = render(<Calculator />);
    expect(myrender).toMatchSnapshot();
  });
});
