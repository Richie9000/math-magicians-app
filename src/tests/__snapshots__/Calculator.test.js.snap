// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Calculator snapshot test 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="calc-container"
      >
        <div
          class="calc-title"
        >
          <p>
            Let's do some  math
          </p>
        </div>
        <div
          class="calculator"
        >
          <div
            class="calculator-display"
          >
            0
          </div>
          <div
            class="calculator-keypad"
          >
            <div
              class="input-keys"
            >
              <div
                class="function-keys"
              >
                <button
                  class="calculator-key key-clear"
                  type="button"
                  value="AC"
                >
                  AC
                </button>
                <button
                  class="calculator-key key-sign"
                  type="button"
                  value="+/-"
                >
                  +/-
                </button>
                <button
                  class="calculator-key key-percent"
                  type="button"
                  value="%"
                >
                  %
                </button>
              </div>
              <div
                class="digit-keys"
              >
                <button
                  class="calculator-key key-0"
                  type="button"
                  value="0"
                >
                  0
                </button>
                <button
                  class="calculator-key key-dot"
                  type="button"
                  value="."
                >
                  ●
                </button>
                <button
                  class="calculator-key key-1"
                  type="button"
                  value="1"
                >
                  1
                </button>
                <button
                  class="calculator-key key-2"
                  type="button"
                  value="2"
                >
                  2
                </button>
                <button
                  class="calculator-key key-3"
                  type="button"
                  value="3"
                >
                  3
                </button>
                <button
                  class="calculator-key key-4"
                  type="button"
                  value="4"
                >
                  4
                </button>
                <button
                  class="calculator-key key-5"
                  type="button"
                  value="5"
                >
                  5
                </button>
                <button
                  class="calculator-key key-6"
                  type="button"
                  value="6"
                >
                  6
                </button>
                <button
                  class="calculator-key key-7"
                  type="button"
                  value="7"
                >
                  7
                </button>
                <button
                  class="calculator-key key-8"
                  type="button"
                  value="8"
                >
                  8
                </button>
                <button
                  class="calculator-key key-9"
                  type="button"
                  value="9"
                >
                  9
                </button>
              </div>
            </div>
            <div
              class="operator-keys"
            >
              <button
                class="calculator-key key-divide"
                type="button"
                value="÷"
              >
                ÷
              </button>
              <button
                class="calculator-key key-multiply"
                type="button"
                value="x"
              >
                *
              </button>
              <button
                class="calculator-key key-substract"
                type="button"
                value="-"
              >
                -
              </button>
              <button
                class="calculator-key key-add"
                type="button"
                value="+"
              >
                +
              </button>
              <button
                class="calculator-key key-equals"
                type="button"
                value="="
              >
                =
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="calc-container"
    >
      <div
        class="calc-title"
      >
        <p>
          Let's do some  math
        </p>
      </div>
      <div
        class="calculator"
      >
        <div
          class="calculator-display"
        >
          0
        </div>
        <div
          class="calculator-keypad"
        >
          <div
            class="input-keys"
          >
            <div
              class="function-keys"
            >
              <button
                class="calculator-key key-clear"
                type="button"
                value="AC"
              >
                AC
              </button>
              <button
                class="calculator-key key-sign"
                type="button"
                value="+/-"
              >
                +/-
              </button>
              <button
                class="calculator-key key-percent"
                type="button"
                value="%"
              >
                %
              </button>
            </div>
            <div
              class="digit-keys"
            >
              <button
                class="calculator-key key-0"
                type="button"
                value="0"
              >
                0
              </button>
              <button
                class="calculator-key key-dot"
                type="button"
                value="."
              >
                ●
              </button>
              <button
                class="calculator-key key-1"
                type="button"
                value="1"
              >
                1
              </button>
              <button
                class="calculator-key key-2"
                type="button"
                value="2"
              >
                2
              </button>
              <button
                class="calculator-key key-3"
                type="button"
                value="3"
              >
                3
              </button>
              <button
                class="calculator-key key-4"
                type="button"
                value="4"
              >
                4
              </button>
              <button
                class="calculator-key key-5"
                type="button"
                value="5"
              >
                5
              </button>
              <button
                class="calculator-key key-6"
                type="button"
                value="6"
              >
                6
              </button>
              <button
                class="calculator-key key-7"
                type="button"
                value="7"
              >
                7
              </button>
              <button
                class="calculator-key key-8"
                type="button"
                value="8"
              >
                8
              </button>
              <button
                class="calculator-key key-9"
                type="button"
                value="9"
              >
                9
              </button>
            </div>
          </div>
          <div
            class="operator-keys"
          >
            <button
              class="calculator-key key-divide"
              type="button"
              value="÷"
            >
              ÷
            </button>
            <button
              class="calculator-key key-multiply"
              type="button"
              value="x"
            >
              *
            </button>
            <button
              class="calculator-key key-substract"
              type="button"
              value="-"
            >
              -
            </button>
            <button
              class="calculator-key key-add"
              type="button"
              value="+"
            >
              +
            </button>
            <button
              class="calculator-key key-equals"
              type="button"
              value="="
            >
              =
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
